File: celladapter/src/main/java/io/erva/celladapter/v7/select/SelectableCell.java
package io.erva.celladapter.v7.select;

import android.view.View;

import io.erva.celladapter.v7.Cell;
import io.erva.celladapter.v7.select.mode.SelectionManager;

public abstract class SelectableCell<ITEM, LISTENER extends Cell.Listener<ITEM>> extends Cell<ITEM, LISTENER> {

    protected SelectionManager selectionManager;

    public SelectableCell(View view) {
        super(view);
    }

    public void setSelectionManager(SelectionManager selectionManager) {
        this.selectionManager = selectionManager;
    }
}

File: celladapter-kotlin/src/main/kotlin/io/erva/celladapter/select/SelectableCell.kt
package io.erva.celladapter.select

import android.view.View

import io.erva.celladapter.Cell
import io.erva.celladapter.select.mode.SelectionManager

abstract class SelectableCell<ITEM : Any, out LISTENER : Cell.Listener<ITEM>>(view: View) : Cell<ITEM, LISTENER>(view) {

    lateinit var selectionManager: SelectionManager
}